src/asm_v_wasm.h

```diff
@@ -29,12 +29,13 @@ AsmType wasmToAsmType(WasmType type);
 
 char getSig(WasmType type);
 
-std::string getSig(FunctionType *type);
+std::string getSig(const FunctionType *type);
 
 std::string getSig(Function *func);
 
-template<typename CallBase>
-std::string getSig(CallBase *call) {
+template<typename T,
+         typename std::enable_if<std::is_base_of<Expression, T>::value>::type* = nullptr>
```

Comment:
It seems like it was always ambiguous with `getSig(FunctionType*)` (i.e. this template could have been instantiated with `CallBase` as `FunctionType` which isn't what we wanted). But it wasn't until this PR that we started seeing errors (i.e. everywhere we pass `FunctionType*` this template is getting instantiated instead) of the function. I'm guess it's because I changed `FunctionType*` to `const FunctionType *` in the other declaration. In any case this makes a bit more explicit the intention that it only work on calls. Unfortunately there is no `CallBase` class, just Expression with 2 kinds of calls, but I think it's still an improvement (even aside from now being necessary).